x-pack/legacy/plugins/reporting/export_types/csv/types.d.ts

```diff
@@ -32,3 +33,73 @@ export type ESQueueCreateJobFnDiscoverCsv = (
   headers: ConditionalHeaders,
   request: Request
 ) => Promise<JobParamsDiscoverCsv>;
+
+type EndpointCaller = (method: string, params: any) => Promise<any>;
+type FormatsMap = Map<
+  string,
+  {
+    id: string;
+    params: {
+      pattern: string;
+    };
+  }
+>;
+
+export interface SearchRequest {
+  index: string;
+  body:
+    | {
+        _source: {
+          excludes: string[];
+          includes: string[];
+        };
+        docvalue_fields: string[];
+        query:
+          | {
+              bool: {
+                filter: any[];
+                must_not: any[];
+                should: any[];
+                must: any[];
+              };
+            }
+          | any;
+        script_fields: any;
+        sort: Array<{
+          [key: string]: {
+            order: string;
+          };
+        }>;
+        stored_fields: string[];
+      }
+    | any;
+}
+
+export interface GenerateCsvParams {
+  searchRequest: SearchRequest;
+  callEndpoint: EndpointCaller;
+  fields: string[];
+  formatsMap: FormatsMap;
+  metaFields: string[]; // FIXME not sure what this is for
+  conflictedTypesFields: string[]; // FIXME not sure what this is for
+  cancellationToken: CancellationToken;
+  settings: {
+    separator: string;
+    quoteValues: boolean;
+    timezone: string | null;
+    maxSizeBytes: number;
+    scroll: { duration: string; size: number };
+  };
+}
+
+export interface SavedSearchGeneratorResult {
+  content: string;
+  csvContainsFormulas: boolean;
+  maxSizeReached: boolean;
+  size: number;
+}
+
+export interface CsvResultFromSearch {
+  type: string;
+  result: SavedSearchGeneratorResult;
+}
```

Comment:
Moved some types to the original `csv` export type typings file, since this should all be common with `csv_from_savedobject` typings.